### PR TITLE
Create gh CI action

### PR DESCRIPTION
**Pull request by remkop22**

### CODE REVIEW

**Review by bheylin — COMMENTED** [review]
I would look at another workflow for inspiration:

https://github.com/dtolnay/anyhow/blob/master/.github/workflows/ci.yml

File: .github/workflows/rust.yml
Comment (bheylin) [inline .github/workflows/rust.yml:1]: Nice! That's clearer to read :wink:

**Review by bheylin — CHANGES_REQUESTED** [review]

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+env:
+  CARGO_TERM_COLOR: always
+  RUSTFLAGS: "-Dwarnings"
+
+jobs: 
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - run: cargo build --verbose
```
[the remark this answers is not in the document]
Comment (remkop22) [reply .github/workflows/rust.yml:18]: According to cargo docs target selection if no package is passed, is based on the current directory, and for virtual workspaces, this includes all members

[the remark this answers is not in the document]
Comment (remkop22) [reply .github/workflows/rust.yml:18]: Hmm, but maybe it's nice to be explicit anyway... I'll change it

**Review by bheylin — APPROVED** [review]

**Pull request merged**